### PR TITLE
Updated `for` to `for..of` 

### PR DESCRIPTION
**Pull request by PrajwalBorkar**
#### Related issues

Part of #18103 
#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by PrajwalBorkar** [comment]
`clients.matchAll(options).then(function(clientList) {
  for (var i = 0 ; i < clientList.length ; i++) {
    if (clientList[i].url === 'index.html') {
      clients.openWindow(clientList[i]);
      // or do something else involving the matching client
    }
  }
});
`
Should I update this code as per this [comment](https://github.com/mdn/content/pull/18485#issuecomment-1188689012)

File: files/en-us/web/api/client/url/index.md [hunk 1]
```diff
@@ -33,8 +33,8 @@ self.addEventListener('notificationclick', function(event) {
   event.waitUntil(clients.matchAll({
     type: 'window'
   }).then(function(clientList) {
-    for (let i = 0; i < clientList.length; i++) {
-      const client = clientList[i];
+    for (const clientListVal of clientList) {
+      const client = clientListVal;
```
Comment (lionralfs) [inline files/en-us/web/api/client/url/index.md:37]:
```suggestion
    for (const client of clientList) {
```

File: files/en-us/web/api/crypto_property/index.md [hunk 2]
```diff
@@ -42,8 +42,8 @@ genRandomNumbers = function getRandomNumbers() {
 
   const randText = document.getElementById("myRandText");
   randText.textContent = "The random numbers are: "
-  for (let i = 0; i < array.length; i++) {
-    randText.textContent += `${array[i]} `;
+  for (const arrayVal of array ) {
```
Comment (lionralfs) [inline files/en-us/web/api/crypto_property/index.md:45]: extra space:

```suggestion
  for (const arrayVal of array) {
```

**Comment by lionralfs** [comment]
> `clients.matchAll(options).then(function(clientList) { for (var i = 0 ; i < clientList.length ; i++) { if (clientList[i].url === 'index.html') { clients.openWindow(clientList[i]); // or do something else involving the matching client } } }); `
> Should I update this code as per this [comment](https://github.com/mdn/content/pull/18485#issuecomment-1188689012)

Looks like you can do:
```js
clients.matchAll(options).then(function(clientList) {
  for (const client of clientList) {
    if (client.url === 'index.html') {
      clients.openWindow(client);
      // or do something else involving the matching client
    }
  }
});
```

File: files/en-us/web/api/crypto_property/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/api/crypto_property/index.md:45]: Also @PrajwalBorkar the `arrayVal` naming looks very artificial. I will prefer `for (const value of array)`.

File: files/en-us/web/api/crypto_property/index.md [hunk 3]
```diff
@@ -36,15 +36,12 @@ This example uses the `crypto` property to access the {{domxref("Crypto.getRando
...
+globalThis.genRandomNumbers = function () {
   const array = new Uint32Array(10);
   crypto.getRandomValues(array);
 
   const randText = document.getElementById("myRandText");
-  randText.textContent = "The random numbers are: "
-  for (let i = 0; i < array.length; i++) {
-    randText.textContent += `${array[i]} `;
-  }
+  randText.textContent = `The random numbers are: ${array.join(" ")}`;
```
Comment (Josh-Cena) [inline files/en-us/web/api/crypto_property/index.md:44]: I've refactored it further. This one needs to be backported to the live examples.

File: files/en-us/web/api/clients/matchall/index.md [hunk 4]
```diff
@@ -56,9 +56,9 @@ order, correct as per spec.
 
 ```js
 clients.matchAll(options).then(function(clientList) {
```
Comment (Josh-Cena) [inline files/en-us/web/api/clients/matchall/index.md:58]:
```suggestion
clients.matchAll(options).then((clientList) => {
```

**Review by jpmedley — APPROVED** [review]

**Pull request merged**